app/pages/admin/project-status.jsx

```diff
@@ -216,6 +227,18 @@ class ProjectStatus extends Component {
                   })}
                 </select>
               </label>
+              {' | '}
+              <label>
+                Retirement:{' '}
+                <select
+                  id='retirementConfig'
+                  onChange={this.onChangeWorkflowRetirement.bind(null, workflow)}
```

Comment:
`onChangeWorkflowRetirement` was already bound in the constructor. If you're binding it on every render here, then the binding in the constructor is redundant.

Out of interest, why does this set the `this` context to null?

Reply:
Alternatively, bind it in the constructor and change this line to call the bound function.
```suggestion
                  onChange={(event) => this.onChangeWorkflowRetirement(workflow, event)}
```

Reply:
I blindly copied it from the current implementation for levels tbh, https://github.com/zooniverse/Panoptes-Front-End/blob/772d5dd2a0307919e072fa9e1cca72458b94d6d5/app/pages/admin/project-status.jsx#L201

Reply:
Shall i update the existing change bind setting as well?

Reply:
Which line are you referring to? My suggested change is to only bind the handler once, in the constructor, then call that bound function whenever the selected option changes.

It needs to be bound to the component because the `this` context of a DOM event is the event target, not the React component instance. If the handler isn't bound, then `this` in the `onChange` handler will reference the `select` element.

Reply:
this one, https://github.com/zooniverse/Panoptes-Front-End/blob/772d5dd2a0307919e072fa9e1cca72458b94d6d5/app/pages/admin/project-status.jsx#L201

Reply:
Yeah, that should probably change too. Otherwise, it's creating a new function on each render, and the old functions don't get properly disposed of.

Reply:
i've updated them both and tested locally with success. Thanks for the suggested change as well.